src/Microsoft.VisualStudio.Editors/PropPages/CodeAnalysisPropPage.vb

```diff
@@ -0,0 +1,246 @@
+        End Function
+
+        ' Returns False if attempt to GetInstalledPackages did not succeed
+        Private Function TryGetInstalledFxCopAnalyzersVersion(ByRef versionString As String) As Boolean
+            versionString = Nothing
+
+            If DTEProject Is Nothing Then
+                Return False
+            End If
+
+            Using session = _threadedWaitDialogFactory.StartWaitDialog(My.Resources.Strings.RestoringPackagesMessage)
+                _packageRestorer.RestorePackages(DTEProject)
+
+                session.UserCancellationToken.ThrowIfCancellationRequested()
+
+                ' Add a small delay to ensure GetInstalledPackages invocation below gets updated packages.
+                Thread.Sleep(2000)
+
+                Dim installedPackages As IEnumerable(Of IVsPackageMetadata)
+                Try
+                    installedPackages = _packageInstallerServices.GetInstalledPackages(DTEProject)
+                Catch
+                    Return False
+                End Try
+
+                For Each package As IVsPackageMetadata In installedPackages
+                    If String.Equals(package.Id, FxCopAnalyzersPackageId, StringComparison.OrdinalIgnoreCase) OrElse
+                       s_childPackageIds.Contains(package.Id, StringComparer.OrdinalIgnoreCase) Then
+                        versionString = package.VersionString
+                        Exit For
+                    End If
+                Next
+
+                Return True
+            End Using
+        End Function
+
+        Private Sub InstallLatestVersionButton_Click(sender As Object, e As EventArgs) Handles InstallLatestVersionButton.Click
+            Debug.Assert(DTEProject IsNot Nothing)
+
+            Try
+                _packageInstaller.InstallPackage(NugetFeed, DTEProject, FxCopAnalyzersPackageId, s_latestStableVersion, ignoreDependencies:=False)
+            Catch
+                _installedFxCopAnalyzersVersionString = Nothing
+            End Try
+
+            RefreshInstalledFxCopAnalyzersVersionAndButtons()
+
+            If _installedFxCopAnalyzersVersionString Is Nothing Then
+                DesignerFramework.DesignUtil.ShowMessage(ServiceProvider, My.Resources.Strings.FxCopAnalyzersInstallFailedMessage, DesignerFramework.DesignUtil.GetDefaultCaption(ServiceProvider), MessageBoxButtons.OK, MessageBoxIcon.Error)
+            End If
+        End Sub
+
+        Private Sub UninstallAnalyzersButton_Click(sender As Object, e As EventArgs) Handles UninstallAnalyzersButton.Click
+            Debug.Assert(DTEProject IsNot Nothing)
+            Debug.Assert(_installedFxCopAnalyzersVersionString IsNot Nothing)
+            Dim versionText = _installedFxCopAnalyzersVersionString
+
+            Try
+                _packageUninstaller.UninstallPackage(DTEProject, FxCopAnalyzersPackageId, removeDependencies:=True)
+            Catch
+            End Try
```

Comment:
Would it be useful to display the error to the user, or some how log it, rather than just show a generic failure message, or is there no extra detail provided? Same question for installation failures I guess.

Reply:
I am hoping this would not happen often, if at all. I can include the message but that might clutter the dialog and may not be really helpful. We can probably wait for feedback on this one..